src/libraries/System.Private.CoreLib/src/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -730,6 +744,11 @@ private bool TryDequeueSlow([MaybeNullWhen(false)] out T item)
                     if (head == _head)
                     {
                         _head = head._nextSegment;
+                        // Attempt to break the chain of segments in order to prevent unbounded memory growth,
+                        // in case one segment remains pinned in memory (which can happen with mono). We can't race
+                        // with the start of an enumeration because it needs to first take the cross segement lock.
+                        if (pendingEnumerations == 0)
+                            head._nextSegment = null;
```

Comment:
ugh. This looks racy with other dequeuers

Reply:
Yes,  this will break dequeues. A dequeuer could have already grabbed the head and needs to be able to fully traverse all linked segments from there.